wordle-web/src/views/WordEditor.vue

```diff
@@ -0,0 +1,152 @@
+            <th class="text-left">Word</th>
+            <th class="text-left">Is Common</th>
+            <th class="text-left">Is Used</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr v-for="word in list" :key="word.wordId">
+            <td>{{ word.text }}</td>
+            <td>
+              <v-checkbox
+                v-model="word.isCommon"
+                v-if="signInService.isSignedIn"
+                @change="updateWord(word)"
+              />
+              <v-icon v-if="!signInService.isSignedIn && word.isCommon" color="green"
+                >mdi-check</v-icon
+              >
+              <v-icon v-if="!signInService.isSignedIn && !word.isCommon">mdi-alpha-x</v-icon>
+            </td>
+            <td>
+              <v-checkbox
+                v-model="word.isUsed"
+                v-if="signInService.isSignedIn"
+                @change="updateWord(word)"
+              />
+              <v-icon v-if="!signInService.isSignedIn && word.isUsed">mdi-check</v-icon>
+              <v-icon v-if="!signInService.isSignedIn && !word.isUsed">mdi-alpha-x</v-icon>
+            </td>
+            <td>
+              <v-icon color="red" @click="deleteWord(word.wordId)">mdi-delete</v-icon>
+            </td>
+          </tr>
+        </tbody>
+      </v-table>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script setup lang="ts">
+import Axios from 'axios'
+import { inject, reactive } from 'vue'
+import { WordListOptions, WordListItem } from '@/scripts/wordEdit'
+import { watch } from 'vue'
+import type { SignInService } from '@/scripts/signInService'
+import { Services } from '@/scripts/services'
+import type { Word } from '@/scripts/word'
+const options = reactive(new WordListOptions())
+const list = reactive(new Array<WordListItem>())
+const signInService = inject(Services.SignInService) as SignInService
+
+watch(options, () => {
+  refreshList()
+})
+refreshList()
+function refreshList() {
+  Axios.get('word/GetWordList', {
+    params: options
+  })
+    .then((response) => {
+      list.length = 0
+      response.data.forEach((item: WordListItem) => {
+        list.push(item)
+      })
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+}
+function deleteWord(id: number) {
+  Axios.delete('word/DeleteWord', {
+    params: {
+      wordId: id
+    }
+  })
+    .then((response) => {
+      console.log(response.data)
+      refreshList()
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+}
+
+function updateWord(word: WordListItem) {
+  console.log(word)
+  Axios.post('word/UpdateWord', word)
+    .then((response) => {
+      console.log(response.data)
+      refreshList()
+    })
+    .catch((error) => {
+      console.log(error)
+    })
+}
+
+function addWord() {
+  // User types in the word in a dialog box
+  const text = prompt('Enter the word to add')
+  if (!text) {
+    return
+  }
+  Axios.post('word/AddWordFromBody', {
+    text: 'text',
```

Comment:
![image](https://github.com/IntelliTect-Samples/EWU-CSCD379-2023-Spring/assets/43832670/a5e1512d-077d-4f07-9ae9-df42b2779e3c)

When adding a word, you aren't sending back the word I want to add, you're sending back the word 'text', so I get an error about not being able to add the word because it's not 5 characters long.